akka-sample-distributed-data-scala/src/main/scala/sample/distributeddata/ShoppingORCart.scala

```diff
@@ -0,0 +1,143 @@
+package sample.distributeddata
+
+import java.util.UUID
+
+import scala.concurrent.duration._
+import akka.actor.Actor
+import akka.actor.ActorRef
+import akka.actor.Props
+import akka.cluster.Cluster
+import akka.cluster.ddata.{DistributedData, ORMultiMap, ORMultiMapKey}
+
+import scala.collection.immutable.HashSet
+
+object ShoppingORCart {
+  import akka.cluster.ddata.Replicator._
+
+  def props(userId: String): Props = Props(new ShoppingORCart(userId))
+
+  case object GetCart
+  final case class ChangeItemQuantity(item: LineItem)
+  final case class RemoveItem(productId: String)
+
+  final case class Cart(items: Set[LineItem])
+  final case class LineItem(productId: String, title: String, quantity: Int)
+  final case class SingleLineItem(productId: String, title: String, uid: String)
+
+  //#read-write-majority-orcart
+  private val timeout = 3.seconds
+  private val readMajority = ReadMajority(timeout)
+  private val writeMajority = WriteMajority(timeout)
+  //#read-write-majority-orcart
+
+  def getUniqueId(implicit cluster: Cluster): String = {
+    cluster.selfUniqueAddress.toString + UUID.randomUUID().toString
+  }
+
+}
+
+class ShoppingORCart(userId: String) extends Actor {
+  import ShoppingORCart._
+  import akka.cluster.ddata.Replicator._
+
+  val replicator = DistributedData(context.system).replicator
+  implicit val cluster = Cluster(context.system)
```

Comment:
new api is `DistributedData(context.system).selfUniqueAddress` instead of implicit Cluster